### ISSUE TITLE
Support for p <= 0?

### ISSUE THREAD

**Issue by kgryte** [compute-io/minkowski-distance#2]
See the [literature](http://www.datatheory.nl/pdfs/95/95_03.pdf).

**Comment by mikolalysenko** [comment]
For p<1, the p-norm doesn't give you a metric.

**Comment by kgryte** [comment]
Indeed. This is something we have debated but not resolved. Different implementations take different stances.
- SciPy [here](https://github.com/scipy/scipy/blob/v0.15.1/scipy/spatial/distance.py#L129) and [here](https://github.com/scipy/scipy/blob/v0.15.1/scipy/linalg/misc.py#L12): while the spatial distance method does not allow for `p<1` and requires the input be an `int`, the `norm` method allows "norms" `<=0`, as it may be useful for certain "numerical" purposes.
- Matlab in their pairwise distance implementation [pdist](https://gene-armada.googlecode.com/svn/trunk/Matlab%20Altered/pdist.m) require only that `p` is a [scalar positive](http://www.mathworks.com/help/stats/pdist.html).

So, the question for us is, while we talk about the module as calculating a distance (metric), whether we would be `ok` with the module being used with alternative `p` values for certain "numerical purposes".

I don't think we have any allegiance here. But we do have the option of drawing a line in the sand, should we desire. I feel like you may have an opinion here.
